examples/tutorials/mvdr_tutorial.py

```diff
@@ -38,258 +52,315 @@
+    reference = reference - reference.mean()
+    reference_pow = reference.pow(2).mean(axis=1, keepdim=True)
+    mix_pow = (estimate * reference).mean(axis=1, keepdim=True)
+    scale = mix_pow / (reference_pow + epsilon)
+
+    reference = scale * reference
+    error = estimate - reference
+
+    reference_pow = reference.pow(2)
+    error_pow = error.pow(2)
+
+    reference_pow = reference_pow.mean(axis=1)
+    error_pow = error_pow.mean(axis=1)
+
+    si_snr = 10 * torch.log10(reference_pow) - 10 * torch.log10(error_pow)
+    return si_snr.item()
 
-for filename in filenames:
-    os.makedirs("_assets", exist_ok=True)
-    if not os.path.exists(filename):
-        with open(f"_assets/{filename}", "wb") as file:
-            file.write(requests.get(f"{base_url}/{filename}").content)
 
 ######################################################################
-# Generate the Ideal Ratio Mask (IRM)
-# -----------------------------------
+# 3. Generate the Ideal Ratio Mask (IRM)
+# --------------------------------------
 #
 
+
 ######################################################################
-# Loading audio data
-# ~~~~~~~~~~~~~~~~~~
+# 3.1. Load audio data
+# ~~~~~~~~~~~~~~~~~~~~
 #
 
-mix, sr = torchaudio.load("_assets/mix.wav")
-reverb_clean, sr2 = torchaudio.load("_assets/reverb_clean.wav")
-clean, sr3 = torchaudio.load("_assets/clean.wav")
-assert sr == sr2
+waveform_mix, sr = torchaudio.load(SAMPLE_MIX)
+waveform_clean, sr2 = torchaudio.load(SAMPLE_CLEAN)
+assert sr == sr2 == SAMPLE_RATE
+# The mixture waveform is a combination of clean and noise waveforms
+waveform_noise = waveform_mix - waveform_clean
 
-noise = mix - reverb_clean
 
 ######################################################################
+# Note: To imrove the robustness of computation, it is recommended to use the
+# double precision (``torch.float64`` or ``torch.double`` for the waveforms.
 #
-# .. note::
-#    The MVDR Module requires ``torch.cdouble`` dtype for noisy STFT.
-#    We need to convert the dtype of the waveforms to ``torch.double``
-#
 
-mix = mix.to(torch.double)
-noise = noise.to(torch.double)
-clean = clean.to(torch.double)
-reverb_clean = reverb_clean.to(torch.double)
+waveform_mix = waveform_mix.to(torch.double)
+waveform_clean = waveform_clean.to(torch.double)
+waveform_noise = waveform_noise.to(torch.double)
+
 
 ######################################################################
-# Compute STFT
-# ~~~~~~~~~~~~
+# 3.2. Compute complex-valued Spectrums
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #
 
+N_FFT = 1024
+N_HOP = 256
 stft = torchaudio.transforms.Spectrogram(
-    n_fft=1024,
-    hop_length=256,
+    n_fft=N_FFT,
+    hop_length=N_HOP,
     power=None,
 )
-istft = torchaudio.transforms.InverseSpectrogram(n_fft=1024, hop_length=256)
+istft = torchaudio.transforms.InverseSpectrogram(n_fft=N_FFT, hop_length=N_HOP)
+
+spectrum_mix = stft(waveform_mix)
+spectrum_clean = stft(waveform_clean)
+spectrum_noise = stft(waveform_noise)
```

Comment:
perhaps rename `spectrum` to `spectrogram` or `spectra`? each result represents a time sequence of spectra, so using the singular "spectrum" is a little misleading